schema/testdata.go

```diff
@@ -242,19 +64,102 @@ func columnsToFields(columns ...Column) []arrow.Field {
 	return fields
 }
 
-// var PKColumnNames = []string{"uuid_pk"}
+func TestTable(name string, opts TestSourceOptions) *Table {
+	t := &Table{
+		Name:    name,
+		Columns: make(ColumnList, 0),
+	}
+	var columns ColumnList
+	// primitive columns
+	columns = append(columns, ColumnList{
+		{Name: "int8", Type: arrow.PrimitiveTypes.Int8},
+		{Name: "int16", Type: arrow.PrimitiveTypes.Int16},
+		{Name: "int32", Type: arrow.PrimitiveTypes.Int32},
+		{Name: "int64", Type: arrow.PrimitiveTypes.Int64},
+		{Name: "uint8", Type: arrow.PrimitiveTypes.Uint8},
+		{Name: "uint16", Type: arrow.PrimitiveTypes.Uint16},
+		{Name: "uint32", Type: arrow.PrimitiveTypes.Uint32},
+		{Name: "uint64", Type: arrow.PrimitiveTypes.Uint64},
+		{Name: "float32", Type: arrow.PrimitiveTypes.Float32},
+		{Name: "float64", Type: arrow.PrimitiveTypes.Float64},
+		{Name: "binary", Type: arrow.BinaryTypes.Binary},
+		{Name: "string", Type: arrow.BinaryTypes.String},
+	}...)
+	if !opts.SkipDates {
+		columns = append(columns, ColumnList{
+			{Name: "date32", Type: arrow.FixedWidthTypes.Date32},
+			{Name: "date64", Type: arrow.FixedWidthTypes.Date64},
+		}...)
+	}
+	if !opts.SkipDurations {
+		columns = append(columns, ColumnList{
+			{Name: "duration_s", Type: arrow.FixedWidthTypes.Duration_s},
+			{Name: "duration_ms", Type: arrow.FixedWidthTypes.Duration_ms},
+			{Name: "duration_us", Type: arrow.FixedWidthTypes.Duration_us},
+			{Name: "duration_ns", Type: arrow.FixedWidthTypes.Duration_ns},
+		}...)
+	}
+
+	if !opts.SkipIntervals {
+		columns = append(columns, ColumnList{
+			{Name: "interval_month", Type: arrow.FixedWidthTypes.MonthInterval},
+			{Name: "interval_day_time", Type: arrow.FixedWidthTypes.DayTimeInterval},
+			{Name: "interval_month_day_nano", Type: arrow.FixedWidthTypes.MonthDayNanoInterval},
+		}...)
+	}
+
+	if !opts.SkipLargeTypes {
+		columns = append(columns, ColumnList{
+			{Name: "large_binary", Type: arrow.BinaryTypes.LargeBinary},
+			{Name: "large_string", Type: arrow.BinaryTypes.LargeString},
+		}...)
+	}
+
+	if !opts.SkipTimes {
+		columns = append(columns, ColumnList{
+			{Name: "time32_s", Type: arrow.FixedWidthTypes.Time32s},
+			{Name: "time32_ms", Type: arrow.FixedWidthTypes.Time32ms},
+			{Name: "time64_us", Type: arrow.FixedWidthTypes.Time64us},
+			{Name: "time64_ns", Type: arrow.FixedWidthTypes.Time64ns},
+		}...)
+	}
+
+	if !opts.SkipTimestamps {
```

Comment:
Theres' an issue here that @hermanschaaf covered in original implementation: we added the `timestamp_us` option always, as this was the default transformed from old cq time.

Reply:
I'll address it in a follow-up PR.

Reply:
It's relatively easy to fix I think, just add `timestamp_us` to the initial list of columns